engines/cryomni3d/detection.h

```diff
@@ -28,7 +28,8 @@
 namespace CryOmni3D {
 
 enum CryOmni3DGameType {
-	GType_VERSAILLES
+	GType_VERSAILLES,
+	GType_DEMO
```

Comment:
```suggestion
	GType_DEMO
```
   GType_VERSAILLES_DEMO

Reply:
`GType_DEMO` is used by Atlantis as well, not just Versailles.

Reply:
Actually... it would be better to use the GF_DEMO game flag for the demo. GType_DEMO is not an actual game

Reply:
As @ccawley2011 implemented if, it is a new subengine so GType_DEMO seems appropriate to me.
If I understood detection code correctly, GF_DEMO would be used if the demo code was implemented in the Versailles engine but as it works for Atlantis demo too that would be strange.

Reply:
If the engine supports other Cryo 3D games in the future, GType_DEMO is too generic. As mentioned, we have a game flag for demos, which is used in other game engines to distinguish demos from full games. With this implementation, GType_DEMO targets both the Versailles and the Atlantis demo, which defeats the purpose of a game type

Reply:
OK. I see what you mean. We can rename the DEMO to GType_HNM4_VIDEO_DEMO because it's what these demos are: a simple HNM4 video player.
What do you think?

Reply:
> OK. I see what you mean. We can rename the DEMO to GType_HNM4_VIDEO_DEMO because it's what these demos are: a simple HNM4 video player. What do you think?

There are two different Atlantis demos: a non-interactive one, and an interactive one, here:
https://archive.org/details/Atlantis_demo

Are these two using the same video format? If yes, would it make sense to use GType_Atlantis for both? Cause again, GType_HNM4_VIDEO_DEMO is not clear which game it refers to, IMHO

Reply:
Interactive demo won't work with this player as I expect it will need a full Atlantis engine before.
So yes it will use GType_Atlantis.

GType_HNM4_VIDEO_DEMO may not mention the HNM version in it as I expect the HNM player to read all HNM files.
It already supports HNM4 and HNM4a, HNM5 (aka UBB, used for Atlantis non-interactive demo) support shouldn't be that hard to add.
HNM6 is quite different (JPEG based) but eventually it could fit in it too.

Do you think GType_HNM_PLAYER would be more appropriate?

Reply:
Good thinking! :)
Yes, GType_HNM_PLAYER is the best option for the non-interactive demo, and GType_Atlantis can be used for the interactive one

Reply:
OK, done.